packages/table/src/headers/_headers.scss

```diff
@@ -215,17 +217,41 @@
   height: $large-cell-height; // fixed height for IE11
 }
 
+.bp-table-column-name,
+.bp-table-row-name {
+  transition: color $cell-transition-duration;
+
+  &.pt-loading {
+    flex: 1;
+  }
+}
+
 .bp-table-column-name {
   display: flex;
   flex-direction: column;
   text-align: left;
   font-size: $pt-font-size;
+
+  &.pt-loading {
+    .bp-table-column-name-text {
+      @include table-skeleton($pt-grid-size * 0.7);
+      @include vertical-center();
+      padding: ($pt-grid-size * 1.1) $pt-grid-size ($pt-grid-size * 1.2);
```

Comment:
the odd height of the table skeleton throws makes the entire shift up and down when transitioning between loading <=> not loading, hence odd padding here